script/deploy/safe/add-safe-owners-and-threshold.ts

```diff
@@ -34,139 +39,184 @@ const main = defineCommand({
       throw new Error(
-        'Private key is missing, either provide it as argument or add PRIVATE_KEY_PRODUCTION to your .env'
+        `Could not get Safe info for ${safeAddress} on ${network}`
       )
+    }
 
-    console.info('Setting up connection to SAFE API')
-
-    const safeService = new SafeApiKit(config)
-
-    const safeAddress = getAddress(networks[network].safeAddress)
-
-    const rpcUrl = chain.rpcUrls.default.http[0] || args.rpcUrl
+    // Get owners from global config and command line arguments
+    let ownersToAdd = [...globalConfig.safeOwners]
 
-    const provider = new ethers.JsonRpcProvider(rpcUrl)
-    const signer = new ethers.Wallet(privateKey, provider)
+    // Add owners from command line if provided
+    if (args.owners) {
+      const cmdLineOwners = args.owners.split(',').map((addr) => addr.trim())
+      consola.info('Adding owners from command line:', cmdLineOwners)
+      ownersToAdd = [...ownersToAdd, ...cmdLineOwners]
+    }
 
-    const ethAdapter = new EthersAdapter({
-      ethers,
-      signerOrProvider: signer,
-    })
+    const currentThreshold = Number(safeInfo.threshold)
 
-    const protocolKit = await Safe.create({
-      ethAdapter,
-      safeAddress: safeAddress,
-      contractNetworks: getSafeUtilityContracts(chain.id),
-    })
+    // Get signer address
+    const senderAddress = safe.account
 
-    const owners = globalConfig.safeOwners
+    consola.info('Safe Address', safeAddress)
+    consola.info('Signer Address', senderAddress)
+    consola.info('Current threshold:', currentThreshold)
+    consola.info('Current owners:', safeInfo.owners)
 
-    let nextNonce = await safeService.getNextNonce(safeAddress)
-    const currentThreshold = (await safeService.getSafeInfo(safeAddress))
-      ?.threshold
-    if (!currentThreshold)
-      throw new Error('Could not get current signature threshold')
+    // Get the next nonce
+    let nextNonce = await getNextNonce(
+      pendingTransactions,
+      safeAddress,
+      network,
+      chain.id,
+      safeInfo.nonce
+    )
 
-    console.info('Safe Address', safeAddress)
-    const senderAddress = await signer.getAddress()
-    console.info('Signer Address', senderAddress)
+    // Go through all owner addresses and add each of them individually
+    for (const o of ownersToAdd) {
+      consola.info('-'.repeat(80))
+      const owner = o as Address
+      const existingOwners = await safe.getOwners()
 
-    // go through all owner addresses and add each of them individually
-    for (const o of owners) {
-      console.info('-'.repeat(80))
-      const owner = getAddress(o)
-      const existingOwners = await protocolKit.getOwners()
-      if (existingOwners.includes(owner)) {
-        console.info('Owner already exists', owner)
+      if (
+        existingOwners.map((o) => o.toLowerCase()).includes(owner.toLowerCase())
+      ) {
+        consola.info('Owner already exists', owner)
```

Comment:
![image](https://github.com/user-attachments/assets/ce88a152-a337-417b-9658-7d274b5dee18)
I am not sure showing this info log.

Reply:
I don't understand?

Reply:
Sorry for not being precise.

the info log message `Owner already exists: {address}` might be misleading. As a user of this script I would assume it refers to the address I just provided even though it's unrelated.

Perhaps adding an additional log message like `Checking owner for {address}` before would tell that it comes from `global.json`.